common/models/create_environment_request.go

```diff
@@ -41,6 +41,26 @@ func (r CreateEnvironmentRequest) Validate() error {
 		return fmt.Errorf("setting `Scale` is not valid for dynamic environments")
 	}
 
+	if r.EnvironmentType == "" {
+		return fmt.Errorf("EnvironmentType is required, please specify 'static' or 'dynamic'")
+	}
+
+	if r.InstanceType == "" {
```

Comment:
InstanceType and UserDataTemplate can be empty if the EnvironmentType is `dynamic`.

Reply:
Okay I was just looking into this. I'll remove that check.

Reply:
Or you can make the check conditional.

Reply:
Something like this?
```
if strings.EqualFold(r.EnvironmentType, EnvironmentTypeStatic) && r.UserDataTemplate == ""
		... error
}
```